sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Processor/PartitionLoadBalancer.cs

```diff
@@ -436,25 +445,38 @@ private async Task RenewOwnershipAsync(CancellationToken cancellationToken)
 
             Logger.RenewOwnershipStart(OwnerIdentifier);
 
-            IEnumerable<EventProcessorPartitionOwnership> ownershipToRenew = InstanceOwnership.Values
+            var utcNow = GetDateTimeOffsetNow();
+
+            List<EventProcessorPartitionOwnership> ownershipToRenew = InstanceOwnership.Values
+                .Where(ownership => (utcNow - ownership.LastModifiedTime) > LoadBalanceInterval)
                 .Select(ownership => new EventProcessorPartitionOwnership
                 {
                     FullyQualifiedNamespace = ownership.FullyQualifiedNamespace,
                     EventHubName = ownership.EventHubName,
                     ConsumerGroup = ownership.ConsumerGroup,
                     OwnerIdentifier = ownership.OwnerIdentifier,
                     PartitionId = ownership.PartitionId,
-                    LastModifiedTime = DateTimeOffset.UtcNow,
+                    LastModifiedTime = utcNow,
                     Version = ownership.Version
-                });
+                })
+                .ToList();
 
             try
             {
-                // Dispose of all previous partition ownership instances and get a whole new dictionary.
+                // Update ownerships we renewed and remove the ones we didn't
```

Comment:
If I'm reading this correctly, I think this changes the behavior.   The previous approach will have instance ownership filled out for all partitions.   In your new approach, are you changing this to only include those that we renew?

Should you, instead, remove just what you renewed and replace them, leaving the other records in-tact?

Reply:
The ClaimOwnershipAsync doesn't return ownership claims that were not renewed. So the old behavior was, effectively, to removing ownership items that failed to renew from the `InstanceOwnership`.

To mirror it in the optimized implementation we need to remove all the items we tried to renew and add the successful back.

Reply:
I see...  I wasn't reading that correctly the first time through.   Now it makes sense.